etna/datasets/tsdataset.py

```diff
@@ -243,7 +268,9 @@ def _check_regressors(df: pd.DataFrame, df_exog: pd.DataFrame):
                     )
 
     def _merge_exog(self, df: pd.DataFrame) -> pd.DataFrame:
-        self._check_regressors(df=df, df_exog=self.df_exog)
+        segments = sorted(set(df.columns.get_level_values("segment")))
+        df_regressors = self.df_exog.loc[:, pd.IndexSlice[segments, self.known_future]]
```

Comment:
Don't you already have this list after _check_known_future call?
Line 114: 
`self._regressors = self._check_known_future(known_future, df_exog)`

Reply:
self._regressors will change according to the new transforms, but we want to check here only initial regressors.

Reply:
Got it